### DISCUSSION TITLE
Inconsistent NextJS Caching on Vercel

### DISCUSSION THREAD

**Discussion by walshhub** [vercel/community#4102] [Help]
### Summary

Hi team,

I'm trying to capitalise on NextJS fetch caching on Vercel, however I need to use an API key. I seem to be getting inconsistent responses when I use headers:

When there's no headers and no API key, the fetch caches fine:
```
export async function getJobs(limit = 10, offset = 0): Promise<JobsResponse> {
  const res = await fetch(
    `https://example.com/api/v1/jobs?limit=${limit}&offset=${offset}`,
    { next: { revalidate: 3600 }}
  );

  return res as JobsResponse;
}
```

When there are headers from the same file, the fetch caches fine here as well:
```
export async function getJobs(limit = 10, offset = 0): Promise<JobsResponse> {
  const someHeaders = { 'X-Auth-Token': process.env.API_KEY } as HeadersInit;

  const res = await fetch(
    `https://example.com/api/v1/jobs?limit=${limit}&offset=${offset}`,
    {headers: someHeaders, next: { revalidate: 3600 }}
  );

  return res as JobsResponse;
}
```

But when I import headers, it doesn't cache at all:
```
// ./constants

export const someHeaders = { 'X-Auth-Token': process.env.API_KEY } as HeadersInit ;
```

```
import { someHeaders } from  ./constants
export async function getJobs(limit = 10, offset = 0): Promise<JobsResponse> {
  const res = await fetch(
    `https://example.com/api/v1/jobs?limit=${limit}&offset=${offset}`,
    {headers: someHeaders, next: { revalidate: 3600 }}
  );

  return res as JobsResponse;
}
```

Any thoughts on how to fix that?

### Steps to Reproduce

To summarise the code above:

1. Do a fetch with no headers and refresh to test caching
2. Do a fetch with headers that are defined in the same function, and refresh to test caching
3. Do a fetch with headers that are defined in a different file, and refresh to test caching.
